code/modules/projectiles/guns/energy/special.dm

```diff
@@ -804,15 +810,19 @@
 	if(linked_pinpointer_UID)
 		if(alert("Do you want to clear the linked pinpointer?", "Pinpointer reset", "Yes", "No") == "Yes")
 			to_chat(user, "<span class='notice'>[src] is ready to be linked to a new pinpointer.</span>")
-			var/obj/item/pinpointer/crew/C = locateUID(linked_pinpointer_UID)
-			C.linked_gun_UID = null
-			if(C.mode == MODE_DET)
-				C.stop_tracking()
-			linked_pinpointer_UID = null
+			unlink()
 
 /obj/item/gun/energy/detective/proc/link_pinpointer(pinpointer_UID)
 	linked_pinpointer_UID = pinpointer_UID
 
+/obj/item/gun/energy/detective/proc/unlink()
+	var/obj/item/pinpointer/crew/C = locateUID(linked_pinpointer_UID)
+	C.linked_gun_UID = null
```

Comment:
```suggestion
	if(!C)
		return
	C.linked_gun_UID = null
```
if your gun get's EMP'd an it's already unlinked, C will be null so you need to perform a sanity check first